packages/lix-sdk/src/branch/merge-branch.ts

```diff
@@ -0,0 +1,171 @@
+import type { Branch, Change } from "../database/schema.js";
+import type { Lix } from "../open/openLix.js";
+import type { DetectedConflict } from "../plugin/lix-plugin.js";
+import { getLowestCommonAncestorV2 } from "../query-utilities/get-lowest-common-ancestor-v2.js";
+
+export async function mergeBranch(args: {
+	lix: Lix;
+	sourceBranch: Branch;
+	targetBranch: Branch;
+}) {
+	const executeInTransaction = async (trx: Lix["db"]) => {
+		const diffingPointers = await getBranchChangePointerDiff(trx, args);
+
+		// could be queried in a single query with a join
+		// but decided to keep it simpler for now
+		const diffingChanges = await trx
+			.selectFrom("change")
+			.select(["id"])
+			.where(
+				"id",
+				"in",
+				diffingPointers.map((pointer) => pointer.source_change_id),
+			)
+			.selectAll()
+			.execute();
+
+		const plugins = await args.lix.plugin.getAll();
+		const detectedConflicts: DetectedConflict[] = [];
+
+		// let plugin detect conflicts
+		await Promise.all(
+			plugins.map(async (plugin) => {
+				if (plugin.detectConflictsV2) {
+					const conflicts = await plugin.detectConflictsV2({
+						lix: args.lix,
+						changes: diffingChanges,
+					});
+					detectedConflicts.push(...conflicts);
+				}
+			}),
+		);
+
+		// update the branch change pointers for non-conflicting changes
+		await Promise.all(
+			diffingPointers.map(async (pointer) => {
+				// potentially expensive query
+				const pluginDetectedConflict = detectedConflicts.find(
+					(conflict) =>
+						conflict.change_id === pointer.target_change_id ||
+						conflict.change_id === pointer.source_change_id,
+				);
+				if (pluginDetectedConflict) {
+					// don't update the branch change pointer
+					return;
+				}
+
+				// if the entity change doesn't exist in the target
+				// it can't conflict (except if a plugin detected
+				// a semantic conflict)
+				const hasConflictingEntityChanges = !pointer.target_change_id
+					? false
+					: await childOfCommonAncestorDiffers({
+							lix: { db: trx },
+							changeA: { id: pointer.source_change_id },
+							changeB: { id: pointer.target_change_id },
+						});
+
+				if (hasConflictingEntityChanges) {
+					detectedConflicts.push({
+						change_id: pointer.source_change_id,
+						conflicting_change_id: pointer.target_change_id!,
+					});
+					return;
+				}
+
+				await trx
+					.insertInto("branch_change_pointer")
+					.values([
+						{
+							branch_id: args.targetBranch.id,
+							change_id: pointer.source_change_id,
+							change_file_id: pointer.change_file_id,
+							change_entity_id: pointer.change_entity_id,
+							change_type: pointer.change_type,
+						},
+					])
+					.onConflict((oc) =>
+						oc.doUpdateSet({ branch_id: args.targetBranch.id }),
+					)
+					.execute();
+			}),
+		);
+
+		// insert the detected conflicts
+		// (ignore if the conflict already exists)
+		if (detectedConflicts.length > 0) {
+			await trx
+				.insertInto("conflict")
```

Comment:
We don't have a relation between conflicts and branches - is this intended?

Reply:
At the moment yes. We need to investigate conflicts in another PR. 

Gut feeling is any branch pointing to the same conflicting change will see the same conflict. If one branch resolves the conflict, it will be automatically in other branches that point to the same change (and thereby conflict).